org.abapgit.adt.ui/src/org/abapgit/adt/ui/internal/wizards/AbapGitWizardPageRepositoryAndCredentials.java

```diff
@@ -306,10 +370,43 @@ private void setUserAndPassControlsVisible(boolean visible) {
 		this.txtPwd.setVisible(visible);
 		this.lblUser.setVisible(visible);
 		this.lblPwd.setVisible(visible);
-		if (!visible) {
-			this.txtUser.setText(""); //$NON-NLS-1$
-			this.txtPwd.setText(""); //$NON-NLS-1$
+	}
+
+	//Retrieve credentials for the given repositoryURL from Secure Store (if they exist)
+	private static IExternalRepositoryInfoRequest getRepoCredentialsFromSecureStorage(String url) {
+		ISecurePreferences preferences = SecurePreferencesFactory.getDefault();
+		String hashedURL = GitCredentialsService.getMD5HashForUrl(url);
+		if (hashedURL != null && preferences.nodeExists(hashedURL)) {
+			ISecurePreferences node = preferences.node(hashedURL);
+			return new IExternalRepositoryInfoRequest() {
+				@Override
+				public String getUser() {
+					try {
+						return node.get("user", null); //$NON-NLS-1$
+					} catch (StorageException e) {
+						AbapGitUIPlugin.getDefault().getLog().log(new Status(IStatus.ERROR, AbapGitUIPlugin.PLUGIN_ID, e.getMessage(), e));
+					}
+					return null;
+				}
+
+				@Override
+				public String getUrl() {
+					return url;
+				}
+
+				@Override
+				public String getPassword() {
+					try {
+						return node.get("password", null); //$NON-NLS-1$
+					} catch (StorageException e) {
+						AbapGitUIPlugin.getDefault().getLog().log(new Status(IStatus.ERROR, AbapGitUIPlugin.PLUGIN_ID, e.getMessage(), e));
+					}
+
+					return null;
+				}
+			};
```

Comment:
This logic retrieves credentials from secure storage and hence can't be public method, which we had decided our discussion.
Since we need to use **getRepoCredentialsFromSecureStorage** in this class, it should be present as a private method here.
Hence, can't move this logic outside.